File: src/main/java/team4/footwithme/team/api/request/TeamCreateRequest.java
package team4.footwithme.team.api.request;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Null;
import team4.footwithme.team.service.request.TeamDefaultServiceRequest;

public record TeamCreateRequest(
        @NotNull(message = "팀 명은 필수입니다.")
        String name,
        @Null
        String description,
        @Null
        String location
) {
    public TeamDefaultServiceRequest toServiceRequest() {
        return new TeamDefaultServiceRequest(
            name, description, location
        );
    }
}

File: src/main/java/team4/footwithme/team/api/request/TeamCreateRequest.kt
package team4.footwithme.team.api.request

import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Null
import team4.footwithme.team.service.request.TeamDefaultServiceRequest

@JvmRecord
data class TeamCreateRequest(
    val name: @NotNull(message = "팀 명은 필수입니다.") String?,
    val description: @Null String?,
    val location: @Null String?
) {
    fun toServiceRequest(): TeamDefaultServiceRequest {
        return TeamDefaultServiceRequest(
            name, description, location
        )
    }
}
